.github/workflows/test.yaml

```diff
@@ -0,0 +1,131 @@
+# Copyright 2020 DigitalOcean
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### Run unit as well as end-to-end tests on PRs and master pushes. For the
+### latter, we also push a new CSI plugin master image to the container
+### registry.
+name: test
+
+env:
+  DOCKER_ORG: digitalocean
+  DOCKER_USER: ${{ secrets.DockerHubUser }}
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
+
+      - name: Go setup
+        uses: actions/setup-go@v1
+        with:
+          go-version: '1.13.5'
+
+      - name: Run unit tests
+        run: make
+
+  push-images:
+    runs-on: ubuntu-18.04
+    needs: unit-test
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@master
+
+      - name: Log into container registry
+        run: echo ${{ secrets.DockerHubToken }} | docker login --username ${DOCKER_USER} --password-stdin
+
+      - name: build and push runner image
+        env:
+          # ${{ github.head_ref }} is empty for non-pull requests, which is
+          # exactly what we want on master.
+          RUNNER_IMAGE_TAG_PREFIX: ${{ github.head_ref }}
+        run: RUNNER_IMAGE=${DOCKER_ORG}/k8s-e2e-test-runner make runner-push
+
+      - name: Build and push plugin image
+        env:
+          VERSION: ${{ github.head_ref }}
+        run: |
+          VERSION="${VERSION:-latest}"
```

Comment:
Should we be defaulting to `master` ?

Reply:
I chose `latest` because it makes the usage of the image slightly easier (you do not need to specify that tag) and it minimizes risk for confusion with the plugin image.

The reason I went with `master` for the plugin image `digitalocean/do-csi-plugin` is that I did not want to give a false impression to users that the latest tag is an alternative name for whatever the latest _stable_ release is (e.g., v1.2.0 as of now), which is usually how `latest` is used.